src/deskolem.ml

```diff
@@ -3,186 +3,216 @@ open! Lplib
 open Term
 open Timed
 
+exception Not_Proof of term
+exception Not_ProofS of term
+
+(** Builtin configuration. *)
+type config =
+  { symb_Skolem     : sym
+  ; symb_Axiom      : sym
+  ; symb_Formula    : sym
+  ; symb_imp        : sym
+  ; symb_forall     : sym
+  ; symb_proof      : sym }
+
+(** [get_config sign] build the configuration using [sign]. *)
+let get_config : Sign.t -> config = fun sign ->
+  let builtin name = Extra.StrMap.find name !(sign.sign_builtins) in
+  { symb_Skolem     = builtin "Skolem" 
+  ; symb_Axiom      = builtin "Axiom" 
+  ; symb_Formula    = builtin "Formula" 
+  ; symb_imp        = builtin "imp"  
+  ; symb_forall     = builtin "forall"
+  ; symb_proof      = builtin "proof" }
+
 (** [subst_inv fu x t] replaces all the subterms [fu] by a fresh variable [x] in
     the term [t]. *)
 let subst_inv : term -> term Bindlib.var -> term -> term = fun fu x ->
   let rec subst t =
     if Eval.eq_modulo [] t fu then mk_Vari x else
     match unfold t with
-    | Vari _                    -> t
-    | Type                      -> t
-    | Kind                      -> t
-    | Symb _                    -> t
-    | Prod( a, b)               ->
+    | Vari _        -> t
+    | Type          -> t
+    | Kind          -> t
+    | Symb _        -> t
+    | Prod( a, b)   ->
         let (v, b) = Bindlib.unbind b in
         let sa = subst a in
         let sb = Bindlib.bind_var v (lift (subst b)) in
-        mk_Prod(sa, Bindlib.unbox sb)
-    | Abst( a, b)               ->
+        mk_Prod (sa, Bindlib.unbox sb)
+    | Abst( a, b)   ->
         let (v, b) = Bindlib.unbind b in
         let sa = subst a in
         let sb = Bindlib.bind_var v (lift (subst b)) in
-        mk_Abst(sa, Bindlib.unbox sb)
-    | Appl( a, b)               -> mk_Appl(subst a, subst b)
+        mk_Abst (sa, Bindlib.unbox sb)
+    | Appl( a, b)   -> mk_Appl (subst a, subst b)
     | Meta _
     | Patt _
     | TEnv _
     | Wild
     | LLet _
-    | TRef _                    -> assert false (* is not handled in the encoding. *)
+    | TRef _        -> assert false (* is not handled in the encoding. *)
   in subst
 
 (** [frozen l t] checks if a term [t] contains one of the variables of [l]. *)
 let frozen : tvar list -> term -> bool = fun l t ->
     let lifted_t = lift t in
+    (* check if all elements of [l] do not occur in [t]. *)
     List.for_all (fun x -> not (Bindlib.occur x lifted_t)) l
 
-(** [print_args l] print terms that [l] contains. *)
-let print_args l =
-    List.iter (Common.Console.out 4 "%a " Print.pp_term) l;
-    Common.Console.out 4 "@."
-
 (** [get_ui f l t] returns a list of arguments used in [f]-terms inside [t]. *)
```

Comment:
More explanations are required here. What is the purpose of each argument?
Also get_ui should be renamed into something more meaningful. Perhaps skolem_args ?